Updated Release Notes. (#551)
Microsoft/mu_plus: Readme.rst

Ubuntu_GCC5_                  |UbuntuCiBuild|   |UbuntuCiTest|  |UbuntuCiCoverage|
============================= ================= =============== ===================

> This repository is part of Project Mu.  Please see Project Mu for details https://microsoft.github.io/mu

Reviewer: Was the period intentionally dropped?

```suggestion
This repository is part of Project Mu.  Please see Project Mu for details https://microsoft.github.io/mu.
```

Branch Status - release/202405
==============================

:Status:
  In Development

:Entered Development:
  2023/11/24

:Anticipated Stabilization:
  Nov 2024

Branch Changes - release/202405
===============================

202405 is a larger deviation than previous releases. As part of upstreaming changes to EDK2, the commits were reviewed, squashed, and some were dropped.

> Due to these changes, there maybe more work minor work required to bring an existing platforms up to 202405 compatibility. 

Reviewer: ```suggestion
Due to these changes, there may be more work required to bring an existing platform up to 202405 compatibility. 
```

Author: I was syncing the ReadMe files between repos, and at least one of the repos does not have the period.
